actor/builtin/miner/miner_test.go

```diff
@@ -677,27 +680,55 @@ func TestMinerSubmitPoStPowerUpdates(t *testing.T) {
 		mal.requireCommit(firstCommitBlockHeight, uint64(1))
 		mal.requireCommit(firstCommitBlockHeight+1, uint64(2))
 		done := types.EmptyIntSet()
-		mal.requirePoSt(firstCommitBlockHeight+5, done)
+		mal.requirePoSt(firstCommitBlockHeight+5, done, faults)
 
 		// Period 2 commit and prove
 		mal.requireCommit(secondProvingPeriodStart+1, uint64(16))
 		mal.requireCommit(secondProvingPeriodStart+2, uint64(17))
-		mal.requirePoSt(secondProvingPeriodStart+5, done)
+		mal.requirePoSt(secondProvingPeriodStart+5, done, faults)
 
 		// Period 3 prove and drop 1 and 2
 		done = types.NewIntSet(1, 2)
-		mal.requirePoSt(thirdProvingPeriodStart+5, done)
+		mal.requirePoSt(thirdProvingPeriodStart+5, done, faults)
 
 		// power lags removal by a proving period
 		power := mal.requirePower(thirdProvingPeriodStart + 6)
 		assert.Equal(t, types.NewBytesAmount(4).Mul(types.OneKiBSectorSize), power)
 
 		// next period power is removed
 		done = types.EmptyIntSet()
-		mal.requirePoSt(fourthProvingPeriodStart+1, done)
+		mal.requirePoSt(fourthProvingPeriodStart+1, done, faults)
 		power = mal.requirePower(fourthProvingPeriodStart + 2)
 		assert.Equal(t, types.NewBytesAmount(2).Mul(types.OneKiBSectorSize), power)
 	})
+
+	t.Run("faults removes power and sector commitments", func(t *testing.T) {
+		mal := setupMinerActorLiason(t)
+
+		// commit several sectors
+		mal.requireCommit(firstCommitBlockHeight, uint64(1))
+		mal.requireCommit(firstCommitBlockHeight, uint64(2))
+		mal.requireCommit(firstCommitBlockHeight, uint64(3))
+
+		done := types.EmptyIntSet()
+
+		// PoSt twice to get power for my sectors
+		mal.requirePoSt(firstCommitBlockHeight+1, done, faults)
+		mal.requirePoSt(firstCommitBlockHeight+1, done, faults)
```

Comment:
Oh this is weird.  Your post should be too early for it to give you power.  cc @acruikshank did we get rid of stopping early PoSt submissions (or did these never make it into the implementation in the first place?).  I am pretty sure we don't want early posts like this because then slight gains can accumulate drastically and trivially allow generation attacks given enough proving periods.  Let's check the spec and file an issue about this.

@ingar this shouldn't get in the way of your PR.  Could you change this test to run the PoSt only once per proving period in the same fashion as the other tests so that once the above correction happens this test is still valid?

Reply:
@acruikshank of course the PoSt construction might change enough in the next month for this to be irrelevant.  I'll leave it up to you to decide if you think this is stable enough to be worth fixing.